README.md

```diff
@@ -14,10 +14,16 @@ This project provides the implementation for Drush site aliases. It is used in D
 
 ### Alias naming conventions
 
-Site alias names always begin with a `@`, and typically are divided in two parts: the site name, and the environment name, each separated by a dot. Neither a site name not an enviornment name may contain a dot. An example alias that referenced the `dev` envionment of the site `example` might therefore look something like:
+Site alias names always begin with a `@`, and typically are divided in three parts: the alias file location (optional), the site name, and the environment name, each separated by a dot. None of these names may contain a dot. An example alias that referenced the `dev` envionment of the site `example` in the `myisp` directory might therefore look something like:
 ```
-@example.dev
+@myisp.example.dev
 ``` 
+The location name is optional. If specified, it will only consider alias files located in directories with the same name as the provided location name. The remainder of the path is immaterial; only the directory that is the immediate parent of the site alias file is relevant. The location name may be omitted, e.g.:
```

Comment:
unless i'm mistaken, this encourages another level of nesting of aliases. but later in the docs we say that no deep searching is done for alias files. so folks must point explicitly to 'schools' dir?

Reply:
Yes, with this PR you must explicitly add every new location to your config directory, or the alias will not be found.

We could consider increasing the search depth by one, but for some locations that might be undesirable. We could consider adding a config location `/path/to/alias/*` to add every directory under a single path. I decided to start with the requirement to add each location explicitly. As you said elsewhere, this is sort of a "Drush for ISPs" feature. Users should only have a handful of ISPs, so unless this feature starts getting used for other purposes, I think the requirement for explicit config is acceptable.

Reply:
Also note that extra locations may be either inside of `sites` or next to it.

Reply:
Actually, the site-alias project allows deep searching -- it has an API to set the depth of the search. The search depth defaults to current-directory only (no deep searching), and Drush does not set it to allow deeper searching.